scripts/helper/src/utils.ts

```diff
@@ -153,3 +152,33 @@ export const getRelayApi = async (endpoint: string): Promise<ApiPromise> => {
     provider: new WsProvider(endpoint)
   })
 }
+
+export const calcWeightPerSecond = (precision: number, price: number): number => {
+  const WEIGHT_PER_SECOND = 10 ** 12
+  // for fixed weigher always 600_000_000
+  const weight = 600_000_000
+  //assume we charge 0.02$ at most for each xcm reserved based transfer
+  const max_fee = 0.02
+  /// max_fee = (weight_per_second * weight)/WEIGHT_PER_SECOND/(10**precision) * price
+  /// so weight_per_second = max_fee*WEIGHT_PER_SECOND*(10**precision)/weight/price
+  const weight_per_second = (((max_fee * WEIGHT_PER_SECOND) / weight) * 10 ** precision) / price
+  /// to avoid price sharply increased later so that we charge too much
+  /// just add some soft limit here
+  return Math.min(1000 * WEIGHT_PER_SECOND, Math.floor(weight_per_second))
+}
```

Comment:
Would like to know if we have any reference documents for this calculation?

Reply:
@MrPai no docs yet but the logic here mainly based on these [xcm configs](https://github.com/parallel-finance/parallel/blob/06be51cb8e9a92fd4f47755917f08434e75727a1/runtime/heiko/src/lib.rs#L1443-L1444)

`Weigher` is to calculate weight consumed for a xcm call and for `polkadotXcm.reservedTransfer` is always `600_000_000`
`Trader` is to calculate how much token equivalent required for these consumed weight and for `FixedRateOfFungible` is just `weight_per_second * weight/WEIGHT_PER_SECOND`

This script is to calculate `weight_per_second` based on that we charge some fixed us dollar(0.02$) for each xcm reserved transfer, so we add token price and precision as param here, we can get token price from some public exchange.

The only tweak I made here is considering currently price for tokens in polkadot ecosystem are somewhat too low, just add some soft limit here to avoid charge too much if price increased later when market gets better.